rust/xaynet-core/src/message/payload/chunk.rs

```diff
@@ -0,0 +1,274 @@
+use std::convert::TryInto;
+
+use anyhow::{anyhow, Context};
+
+use crate::message::{
+    traits::{FromBytes, ToBytes},
+    DecodeError,
+};
+
+pub(crate) mod ranges {
+    use crate::message::utils::range;
+    use std::ops::Range;
+
+    /// Byte range corresponding to the chunk ID in a chunk message
+    pub const ID: Range<usize> = range(0, 2);
+    /// Byte range corresponding to the message ID in a chunk message
+    pub const MESSAGE_ID: Range<usize> = range(ID.end, 2);
+    /// Byte range corresponding to the flags in a chunk message
+    pub const FLAGS: usize = MESSAGE_ID.end;
+    /// Byte range reserved for future use
+    pub const RESERVED: Range<usize> = range(FLAGS + 1, 3);
+}
+
+/// Length in bytes of a chunk message header
+const HEADER_LENGTH: usize = ranges::RESERVED.end;
+
+/// A message chunk.
+#[derive(Eq, PartialEq, Debug, Clone)]
+pub struct Chunk {
+    /// Chunk ID
+    id: u16,
+    /// ID of the message this chunk belongs to
+    message_id: u16,
+    /// `true` if this is the last chunk of the message, `false` otherwise
+    last: bool,
```

Comment:
Is there a way to limit the amount of chunks a participant can send?

Reply:
No. With this the server doesn't know the full message size. However we still have 24 "reserved" bytes, so we could maybe have a 16 bits long field for the total number of chunks. We'd still have 8 bits left.